robota_core/tests/test_artefact_urls.py

```diff
@@ -0,0 +1,16 @@
+import pytest
+from robota_core.repository import GitlabRepository, GithubRepository
+
+class TestGithubUrls:
+    repo = "https://github.com/robota-suite/robota-core"
+    project = "robota-suite/robota-core"
+
+    def test_branch_url(self):
```

Comment:
Sadly, both Github and Gitlab require an API token for anything more complicated than reading off branch names.

I don't see any easy solutions. What do you think?

Reply:
Thanks so much for getting back to me on this, after such a long time, @MeltedHyperion7 .

Is a token needed to access the compare diff through the REST API on a public project?  The test projects I set up are public.  I tried to access the diff from the test GitLab repo (mentioned later in the review) in an incognito browser window and got a non-empty JSON result back.  Does the API call not work for you?

https://gitlab.com/api/v4/projects/60747754/repository/compare?from=INITIAL_COMMIT&to=main

I tried the same thing with GitHub and was able to see the JSON result:

https://api.github.com/repos/robota-suite/test-public-repo/compare/INITIAL_COMMIT...main

But maybe you're telling me that the Python libraries we're using to access these APIs need a token?

Reply:
Thank you for checking this. I think it fails because the code is currently set up to attempt authentication every time and [fails if no token is provided](https://github.com/robota-suite/robota-core/blob/develop/robota_core/gitlab_tools.py#L48).

Should I open an issue to rework the code to work without a token?

Reply:
Oh - right!  So it's a robota-core thing!  Thanks for pointing this out.

Yes - we do want robota-core to be useful for analysing public repositories, so making an issue that requests that feature would be great.

Reply:
(I should have said that we can still write the test for the diff, as we can add a config file with the tokens needed to authenticate, with the secrets being set locally to run the tests in the development environment and through a CI secret for running the tests through GH Actions.)